### DISCUSSION TITLE
textures and meshes in extra projects folder

### DISCUSSION THREAD

**Discussion by Thomas-Feldmeier** [cyberbotics/webots#3944] [Ideas]
[file hierachy](https://cyberbotics.com/doc/guide/the-standard-file-hierarchy-of-a-project)

A project should have these folders:
* worlds
* protos
* controllers
* plugins
* textures (new)
* meshes (new)

This would help to seperate .proto and .wbt and .png/.jpg and .obj files. 

Textures or meshes are searched in /textures/ or /meshes/ subfolder. Maybe in the future they are not searched in protos or worlds subfolder any more. This would make it less complicated:

https://www.cyberbotics.com/doc/reference/imagetexture#search-rule-of-the-texture-path

see [#3922](https://github.com/cyberbotics/webots/discussions/3922)
see [#3942](https://github.com/cyberbotics/webots/issues/3942)

**Comment by omichel** [comment]
We have a plan to simplify this rule, see #1638.
